assembl/static2/js/app/components/debate/brightMirror/brightMirrorInstructionView.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+
+import { Col } from 'react-bootstrap';
+import activeHtml from 'react-active-html';
+import ARange from 'annotator_range'; // eslint-disable-line
+import Embed from '../../common/urlPreview/embed';
+import URLMetadataLoader from '../../common/urlPreview/urlMetadataLoader';
+import { isSpecialURL } from '../../../utils/urlPreview';
+
+
+const postBodyReplacementComponents = () => ({
+  iframe: attributes => (
+    // the src iframe url is different from the resource url
+    <Embed url={attributes['data-source-url'] || attributes.src} defaultEmbed={<iframe title="post-embed" {...attributes} />} />
+  ),
+  a: (attributes) => {
+    const embeddedUrl = isSpecialURL(attributes.href);
+    const origin = (
+      <a key={`url-link-${attributes.href}`} href={attributes.href} className="linkified" target="_blank">
+        {attributes.href}
+      </a>
+    );
+    if (embeddedUrl) return origin;
+    return [origin, <URLMetadataLoader key={`url-preview-${attributes.href}`} url={attributes.href} />];
+  },
+});
+
+
+class BrightMirrorInstructionView extends Component<> {
```

Comment:
this can be a functional component right ?

Reply:
yes +1